app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -108,3 +226,97 @@ const parseUserSession = (session: cognito.CognitoUserSession): UserSession => {
     const accessToken = session.getAccessToken().getJwtToken()
     return { email, accessToken }
 }
+
+// ==============
+// === SignUp ===
+// ==============
+
+const signUp = (username: string, password: string, platform: app.Platform) =>
+    results.Result.wrapAsync(() => {
+        const params = intoSignUpParams(username, password, platform)
+        return amplify.Auth.signUp(params)
+    })
+        /** The contents of a successful response are not relevant, so we discard them. */
+        .then(result => result.map(() => null))
+        .then(result => result.mapErr(intoAmplifyErrorOrThrow))
+        .then(result => result.mapErr(intoSignUpErrorOrThrow))
+
+const intoSignUpParams = (
+    username: string,
+    password: string,
+    platform: app.Platform
+): amplify.SignUpParams => ({
+    username,
+    password,
+    attributes: {
+        email: username,
+        /** Add a custom attribute indicating whether the user is signing up from the desktop. This
+         * is used to determine the schema used in the callback links sent in the verification
+         * emails. For example, `http://` for the Cloud, and `enso://` for the desktop. */
+        /** # Safety
+         *
+         * It is necessary to disable the naming convention rule here, because the key is expected
+         * to appear exactly as-is in Cognito, so we must match it. */
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        'custom:fromDesktop': platform === app.Platform.desktop ? 'true' : 'false',
+    },
+})
+
+type SignUpErrorKind =
```

Comment:
if using the current type, consider switcthing to `typeof KNOWN_ERRORS.signUp[typeof KNOWN_ERRORS.signUp]["kind"]`

if changing to the proposed simpler type, consider moving this next to `KNOWN_ERRORS` so it's harder to forget about updating these

Reply:
Changing to the proposed simpler type.

Reply:
@somebody1234 is this thread solved?